File: src/test/java/nz/co/searchwellington/model/UrlWordsGeneratorTest.java
package nz.co.searchwellington.model;

import static org.junit.Assert.assertEquals;
import nz.co.searchwellington.model.frontend.FrontendNewsitem;

import org.joda.time.DateTime;
import org.joda.time.DateTimeZone;
import org.junit.Before;
import org.junit.Test;

public class UrlWordsGeneratorTest {

	private UrlWordsGenerator urlWordsGenerator;
	
	private FrontendNewsitem newsitem;
	
	@Before
	public void setup() {
		newsitem = new FrontendNewsitem();
		newsitem.setName("Some thing happening");
		DateTime pubdate = new DateTime(2010, 4, 2, 0, 0, 0, 0, DateTimeZone.UTC);
		newsitem.setDate(pubdate.toDate());
		this.urlWordsGenerator = new UrlWordsGenerator();
	}
	
	@Test
	public void shouldProduceCorrectUrlBasedOnPublisherDateAndHeadline() throws Exception {		
		Website publisher = new WebsiteImpl();
		publisher.setName("Island Bay school");
		newsitem.setPublisherName("Island Bay school");
		
		assertEquals("/island-bay-school/2010/apr/2/some-thing-happening", urlWordsGenerator.makeUrlForNewsitem(newsitem));
	}
	
	@Test
	public void urlWordshouldBeDateAndHeadlineIfPublisherIsNotSet() throws Exception {
		assertEquals("/2010/apr/2/some-thing-happening", urlWordsGenerator.makeUrlForNewsitem(newsitem));
	}
	
}

File: src/test/scala/nz/co/searchwellington/model/UrlWordsGeneratorTest.scala
package nz.co.searchwellington.model

import org.joda.time.{DateTime, DateTimeZone}
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import uk.co.eelpieconsulting.common.dates.DateFormatter

class UrlWordsGeneratorTest {

  private val april2010 = new DateTime(2010, 4, 2, 12, 0).withZone(DateTimeZone.UTC).toDate

  private val urlWordsGenerator = new UrlWordsGenerator(new DateFormatter(DateTimeZone.UTC))

  @Test
  def shouldProduceSeoStyleUrlWordsForPublisher(): Unit = {
    val publisher = Website(title = "Island Bay school")

    val urlWords = urlWordsGenerator.makeUrlWordsFor(publisher, None)

    assertEquals("island-bay-school", urlWords)
  }

  @Test
  def shouldProduceCorrectUrlBasedOnPublisherDateAndHeadline(): Unit = {
    val publisher = Website(title = "Island Bay school")
    val newsitemWithPublisher = Newsitem(title = "Something happened at the school", date = april2010, publisher = Some(publisher._id))

    val urlWords = urlWordsGenerator.makeUrlWordsFor(newsitemWithPublisher, Some(publisher))

    assertEquals("island-bay-school/2010/apr/2/something-happened-at-the-school", urlWords)  // TODO should URLS words be slash prefixed?
  }

  @Test
  def urlWordshouldBeDateAndHeadlineIfPublisherIsNotSet(): Unit = {
    val newsitemWithNoPublisher = Newsitem(title = "Something-happening", date = april2010)

    val urlWords = urlWordsGenerator.makeUrlWordsFor(newsitemWithNoPublisher)

    assertEquals("2010/apr/2/something-happening", urlWords)
  }

}
